src/WebJobs.Script/Workers/ProcessManagement/DefaultWorkerProcessFactory.cs

```diff
@@ -89,5 +96,22 @@ internal string SanitizeExpandedArgument(string envExpandedString)
             }
             return envExpandedString;
         }
+
+        internal void RemoveConcurrencyLimits(ProcessStartInfo startInfo)
```

Comment:
Should this instead be named ApplyWorkerConcurrencyLimits?

Reply:
Also, perhaps rather than overriding these environment variables, the workers should take the concurrency options and do this themselves internally, rather than us setting this from the outside?

Reply:
We decided that only function host decides when adjust the level of concurrency. @AnatoliB, what do you think?

Reply:
I like the the word "Remove" a bit more here. Conceptually, removing the limits is the actual intent. Technically, passing very high values is the way to achieve this goal, but this is just an implementation detail (simply because this is the only knob the workers have, there is no way to say "no limit").

We could also pass a boolean flag, like @mathewc is proposing. This will also work, but I can see two problems with this approach:

1. Worker concurrency will now be controlled by _two_ config values instead of one, and the relationship between them is not absolutely obvious. We would have to make sure every language worker (currently - at least Python and PowerShell) handles this pair properly.
2. In order to implement this logic, we'll have to rebuild and re-release every language worker. If we ever want to change this logic, we'll have to rebuild and re-release every language worker again.

With this PR, the situation remains very simple: every worker has a single and trivial concurrency limit knob, and the host is the only component making decisions, so all the relevant changes will be in the host. This will make experimentation a bit easier now, and this is the main reason I like it.

Reply:
For me having function host as only point for making worker concurrency decisions is better as it simplifies testing and we do not need to change the workers. We can change it in future if needed.